src/homeWork17/home_work17.ts

```diff
@@ -0,0 +1,348 @@
+  }
+}
+
+class Reporting {
+    private dailyReport:number = 0;
+
+    collectingDataRevenue(revenue:number):void{
+        this.dailyReport += revenue;
+        console.log(`Report collected ${revenue}`);
+    }
+
+    transferingToAccounting():void{
+        console.log(`Transferring daily revenue to Accounting:${this.dailyReport}`);
+    }
+
+}
+
+class Accountings{
+    private budgetAll:number = 0;
+    private animalExpenses:number = 0;
+    private employeeExpenses:number = 0;
+    private zooMaintenance:number = 0;
+
+    constructor(budget:number){
+        this.budgetAll = budget;
+    }
+
+    addExpenseForAnimal(amount:number):void{
+        this.animalExpenses += amount;
+        console.log(`Expenses of ${amount} added for employee payment`);
+    }
+      
+    addExpenseForEmployee(amount: number):void {
+        this.employeeExpenses += amount;
+        console.log(`Expense of ${amount} added for employee payment.`);
+    }
+    
+    addForZooMaimnenance(amount:number):void{
+        this.zooMaintenance += amount;
+        console.log(`Was spent on maintenance of the zoo`)
+
+    }
+
+  
+    generateFinancialReports():void {
+    console.log("Financial Report:");
+    console.log(`Total animal care expenses: $${this.animalExpenses}`);
+    console.log(`Total employee payment expenses: $${this.employeeExpenses}`);
+    console.log(`Total maintenance of the zoo $${this.zooMaintenance}`)
+    const totalExpenses = this.animalExpenses + this.employeeExpenses + this.zooMaintenance;
+    console.log(`Total expenses: $${totalExpenses}`);
+    const remainingBudget = this.budgetAll - totalExpenses;
+    console.log(`Remaining budget: $${remainingBudget}`);
+    }
+
+}
+
+class Administration{
+    private employes:IEmployees[] = [];
+    private animals:IAnimals[] = [];
+    addEmployes(employes:IEmployees):void{
+        this.employes.push(employes);
+        console.log(`New employe ${employes.name} added`);
+    }
+    deleteEmployes(employesName:string):void{
+        const index = this.employes.findIndex(employee=>employee.name === employesName);
+        if(index !==-1){
+            const deleteEmployes = this.employes.splice(index,1)[0];
+            console.log(`Employes ${deleteEmployes.name} removed succesfully`);
+        }else {
+            console.log(`Employes ${employesName} not found`);
+        }
+    }
+    addAnimals(animals:IAnimals):void{
+        this.animals.push(animals);
+        console.log(`Animal ${animals.name} added succesfully.`);
+    }
+    deleteAnimals(animalsName:string):void{
+        const index = this.animals.findIndex(animals=>animals.name === animalsName);
+        if(index !== -1){
+            const deleteAnimals =this.animals.splice(index, 1)[0];
+            console.log(`Animal ${deleteAnimals.name} delete succesfully.`);
+        }else{
+            console.log(`Animal ${animalsName} not found.`)
+        }
+    }
+    promoNotification(promo:string):void{
```

Comment:
It has come to my attention that both the "Administration" and "MarketingDepartments" classes share the same actions. This could lead to confusion and inefficiencies in our operations. We need to come up with a strategy to effectively manage this situation.